src/components/table/table.e2e.ts

```diff
@@ -0,0 +1,138 @@
+import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
```

Comment:
@adrianschmidt Thanks for questioning my vague description!! I meant it from the perspective of the test, not the component itself. The change needed in the component itself was not making the prop mutable from inside. It was the missing watcher that was required since the test requires to mutate the element after attaching it to the DOM.

This is required for complex properties that can't be passed in the DOM, such as the array of column definitions. It is however not really needed for the selectable property, but I discovered the missing watcher by writing my test and using `setProperty` after adding the element to the DOM. 

I hope this explains it, please let me elaborate more on this or make changes if you feel it requires more clarifications.

Reply:
Ah, yeah, I understand now! Good to hear 👍

I think it's generally better to allow properties to be changed from the outside after the component was added. I think we've had problems when using components in Angular, because even when the property is added to the template code from the beginning, the way Angular creates the component means the properties are updated after the component is created (if I recall correctly). (Frankly, I don't quite understand how Stencil does it. I suspect they might be "cheating" in some way that isn't available to other template engines…)